modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/nhttp/config/ProxyConfigBuilder.java

```diff
@@ -1,82 +1,261 @@
+     *      </profile>
+     *      <profile>
+     *          <targetHosts>*</targetHosts>
+     *          <proxyHost>localhost</proxyHost>
+     *          <proxyPort>7443</proxyPort>
+     *          <bypass>test.com, direct.com</bypass>
+     *      </profile>
+     * </parameter>
+     * }
+     *
+     * @param transportOut transport out description
+     * @return map of <code>ProxyProfileConfig<code/> if configured in axis2.xml; otherwise null
+     * @throws AxisFault if proxy profile is not properly configured
+     */
+    private Map<String, ProxyProfileConfig> getProxyProfiles(TransportOutDescription transportOut) throws AxisFault {
+        Parameter proxyProfilesParam = transportOut.getParameter("proxyProfiles");
+        if (proxyProfilesParam == null) {
+            return null;
+        }
 
-            String s = null;
-            Parameter bypassListParam = transportOut.getParameter("http.nonProxyHosts");
-            if (bypassListParam != null) {
-                s = (String) bypassListParam.getValue();
-            }
-            if (s == null) {
-                s = System.getProperty("http.nonProxyHosts");
+        if (log.isDebugEnabled()) {
+            log.debug(name + " Loading proxy profiles for the HTTP/S sender");
+        }
+
+        OMElement proxyProfilesParamEle = proxyProfilesParam.getParameterElement();
+        Iterator<?> profiles = proxyProfilesParamEle.getChildrenWithName(Q_PROFILE);
+        Map<String, ProxyProfileConfig> proxyProfileMap = new HashMap<String, ProxyProfileConfig>();
+
+        while (profiles.hasNext()) {
+            OMElement profile = (OMElement) profiles.next();
+            OMElement targetHostsEle = profile.getFirstChildWithName(Q_TARGET_HOSTS);
+            if (targetHostsEle == null || targetHostsEle.getText().isEmpty()) {
+                String msg = "Each proxy profile must define at least one host " +
+                        "or a wildcard matcher under the targetHosts element";
+                log.error(name + " " + msg);
+                throw new AxisFault(msg);
             }
-            if (s != null) {
-                proxyBypass = s.split("\\|");
+
+            HttpHost proxy = getHttpProxy(profile, targetHostsEle.getText());
+
+            UsernamePasswordCredentials proxyCredentials = getUsernamePasswordCredentials(profile);
+
+            Set<String> proxyBypass = getProxyBypass(profile);
+
+            ProxyProfileConfig proxyProfileConfig = new ProxyProfileConfig(proxy, proxyCredentials, proxyBypass);
+
+            String[] targetHosts = targetHostsEle.getText().split(",");
+
+            for (String endPoint : targetHosts) {
+                endPoint = endPoint.trim();
+                if (!proxyProfileMap.containsKey(endPoint)) {
+                    proxyProfileMap.put(endPoint, proxyProfileConfig);
+                } else {
+                    log.warn(name + " Multiple proxy profiles were found for the endPoint : " +
```

Comment:
* Yes, "endpoint" is a single word. I'll change as suggested
*  I'll revisit all the log messages and correct them 

Thank you